libraries/libfc/include/fc/io/random_access_file.hpp

```diff
@@ -124,14 +124,17 @@ struct random_access_file_context {
 
          ssize_t wrote = -1;
          do {
-            if(offs == append_t)
+            if(offs == append_t) {
 #ifdef __linux__
                wrote = pwritev2(fd, iov, i, 0, RWF_APPEND);   //linux *not* opened with O_APPEND, appending requires special flag
+               if(wrote == -1 && errno == EOPNOTSUPP)         //fallback for kernels before 4.16
+                  wrote = pwritev(fd, iov, i, size());
```

Comment:
`pwritev2` ignores offset (except -1) when given `RWF_APPEND`.
`pwritev` you need size because you want to append. I agree it is cleaner and more concise. Is it actually more performant?

I wonder if `pwritev` should just be used.

Reply:
Oh, never mind. It is more performant because you don't have to call `size()`.